app/server/src/main/scala/org/bitcoins/server/BitcoindRpcBackendUtil.scala

```diff
@@ -308,56 +308,105 @@ object BitcoindRpcBackendUtil extends Logging {
+                    err)
                 }
-                .toMat(Sink.seq)(Keep.right)
-                .run()
-
-              val requestsBlocksF = for {
-                hashes <- hashFs
-                _ <- wallet.nodeApi.downloadBlocks(hashes.toVector)
-              } yield logger.debug(
-                "Successfully polled bitcoind for new blocks")
-
-              requestsBlocksF.failed.foreach { case err =>
-                val failedCount = atomicPrevCount.get
-                atomicPrevCount.set(prevCount)
-                logger.error(
-                  s"Requesting blocks from bitcoind polling failed, range=[$prevCount, $failedCount]",
-                  err)
-              }
 
-              requestsBlocksF
-            } else if (prevCount > count) {
-              Future.failed(new RuntimeException(
-                s"Bitcoind is at a block height ($count) before the wallet's ($prevCount)"))
-            } else {
-              logger.debug(s"In sync $prevCount count=$count")
-              Future.unit
+                requestsBlocksF
+              } else if (prevCount > count) {
+                Future.failed(new RuntimeException(
+                  s"Bitcoind is at a block height ($count) before the wallet's ($prevCount)"))
+              } else {
+                logger.debug(s"In sync $prevCount count=$count")
+                Future.unit
+              }
             }
+            res.onComplete(_ => processing.set(false))
           }
-          ()
         }
       }
     }
 
     resultF
   }
+
+  def startBitcoindMempoolPolling(
+      bitcoind: BitcoindRpcClient,
+      interval: FiniteDuration = 10.seconds)(
+      processTx: Transaction => Future[Unit])(implicit
+      system: ActorSystem,
+      ec: ExecutionContext): Cancellable = {
+    @volatile var prevMempool: Set[DoubleSha256DigestBE] =
+      Set.empty[DoubleSha256DigestBE]
+
+    def getDiffAndReplace(
+        newMempool: Set[DoubleSha256DigestBE]): Set[DoubleSha256DigestBE] = {
+      val txids = newMempool.diff(prevMempool)
+      prevMempool = newMempool
+      txids
+    }
+
+    val processing = new AtomicBoolean(false)
+
+    system.scheduler.scheduleWithFixedDelay(0.seconds, interval) { () =>
+      {
+        if (processing.compareAndSet(false, true)) {
+          logger.info("Polling bitcoind for mempool")
+
+          val res = for {
+            mempool <- bitcoind.getRawMemPool
+            newTxIds = getDiffAndReplace(mempool.toSet)
+            _ = logger.info(s"Found ${newTxIds.size} new mempool transactions")
+            newTxsOpt <- FutureUtil.sequentially(newTxIds)(
+              bitcoind.getRawTransactionRaw(_).map(Option(_)).recover { _ =>
+                None
+              })
+            newTxs = newTxsOpt.collect { case Some(tx) =>
+              tx
+            }
+            _ <- FutureUtil.sequentially(newTxs)(processTx)
+          } yield {
+            logger.info(
```

Comment:
Perhaps bump this down to `debug`, otherwise this will log every 10 seconds.

It also might be useful to have a log in the case where we are skipping the scheduled job due to the previous run still running? I.e. 

```scala
if (processing.compareAndSet(false, true)) {
   ...
} else { 
  logger.info(s"Skipping scanning the mempool since a previously scheduled task is still running")
}
```

That way we can get some information if for whatever reason the previous run is taking a long time. Probably should do the same thing on `startBitcoindBlockPolling`